src/cocotb/share/lib/embed/gpi_embed.cpp

```diff
@@ -140,11 +130,71 @@ extern "C" COCOTB_EXPORT void _embed_init_python(void) {
     }
 
     to_python();
+
     // must set program name to Python executable before initialization, so
     // initialization can determine path from executable
-    set_program_name_in_venv();
-    Py_Initialize(); /* Initialize the interpreter */
+
+    static wchar_t interpreter_path[PATH_MAX], sys_executable[PATH_MAX];
+
+    if (get_interpreter_path(interpreter_path, sizeof(interpreter_path))) {
+        return;
+    }
+    LOG_INFO("Using Python interpreter at %ls", interpreter_path);
+
+#if PY_VERSION_HEX >= 0x3080000
+    /* Use the new Python Initialization Configuration from Python 3.8. */
+    PyConfig config;
+    PyStatus status;
+
+    PyConfig_InitPythonConfig(&config);
+    PyConfig_SetString(&config, &config.program_name, interpreter_path);
```

Comment:
```suggestion
    DEFER(PyConfig_Clear(&config));

    PyConfig_SetString(&config, &config.program_name, interpreter_path);
```

Then you can remove all the calls below. This is prefered.